server/repository/repository.go

```diff
@@ -159,12 +159,15 @@ func repoToStringData(r *appsv1.Repository) map[string]string {
 	}
 }
 
-// secretToRepo converts a secret into a repository object
-func secretToRepo(s *apiv1.Secret) *appsv1.Repository {
-	return &appsv1.Repository{
-		Repo:          string(s.Data["repository"]),
-		Username:      string(s.Data["username"]),
-		Password:      string(s.Data["password"]),
-		SSHPrivateKey: string(s.Data["sshPrivateKey"]),
+// secretToRepo converts a secret into a repository object, optionally redacting sensitive information
+func secretToRepo(s *apiv1.Secret, redact bool) *appsv1.Repository {
+	repo := appsv1.Repository{
+		Repo: string(s.Data["repository"]),
 	}
+	if !redact {
+		repo.Username = string(s.Data["username"])
```

Comment:
I don't think we necessarily need to redact username. It may be useful to be used as a reference when inputing the repo, and shown in the CLI/UI. Note that with github token authentication (which everyone should be using instead of personal username/password), username is irrelevant and github accepts anything as the username.